main.tf

```
@@ -594,3 +604,176 @@ resource "aws_scheduler_schedule" "this" {
     }
   }
 }
+
+resource "aws_pipes_pipe" "this" {
+  for_each = { for k, v in local.eventbridge_pipes : v.name => v if var.create && var.create_pipes }
+
+  name        = each.value.Name
+  name_prefix = lookup(each.value, "name_prefix", null)
+
+  role_arn = try(each.value.role_arn, aws_iam_role.eventbridge_pipe[each.key].arn)
```

since this is an ARN and therefore a computed value, better to use `lookup()` in case users are wrapping/looping over this module

```suggestion
  role_arn = lookup(each.value, "role_arn", aws_iam_role.eventbridge_pipe[each.key].arn)
```

Ref: https://github.com/clowdhaus/terraform-for-each-unknown

---

I agree that `lookup()` is better (in general) for such computed values, but here it doesn't work because default value (`aws_iam_role.eventbridge_pipe[each.key].arn`) maybe not exist.

The right code is ugly (and `irrelevant` value is used just to make `terraform plan` happy):
```
lookup(each.value, "role_arn", try(aws_iam_role.eventbridge_pipe[each.key].arn, "irrelevant"))
```

I will stick with `try()` since it always works.